scripts/openshiftci-periodic-tests.sh

```diff
@@ -5,13 +5,46 @@ set -e
 # show commands
 set -x
 
+# check if devfiles are updated and if so, update and test them
+check_and_run_devfileTest() {
+    NOTEQUAL="false"
+    # Languages for which devfiles preset in examples dir
+    LANGUAGES=('python' 'nodejs' 'springboot')
+
+    for LANGUAGE in "${LANGUAGES[@]}"; do
+        Example_devfile_path=./tests/examples/source/devfiles/$LANGUAGE/devfile-registry.yaml
+        TEMPDIR=$(mktemp -d)
+        # download devfiles with odo
+        if [[ $LANGUAGE == "springboot" ]]; then
+            odo create java-$LANGUAGE language --context $TEMPDIR
+        else
+            odo create $LANGUAGE language --context $TEMPDIR
+        fi
+
+        Devfile_path=$TEMPDIR/devfile.yaml
+        set +e
```

Comment:
Do we need this?
`false || NOTEQUAL="true"` is not a failing command

You would need it to do smth similar to:
```
set +e
diff $Devfile_path $Example_devfile_path 
set -e
if [[ $? == 1 ]]; then
 ...
fi
```

but with the `||`, you are "catching" the error and this is the result of `NOTEQUAL="true" which applies, which is always OK

Reply:
Updated!